plugin/plugin.go

```diff
@@ -137,6 +139,9 @@ func Register(r *Registration) {
 		panic(ErrInvalidRequires)
 	}
 
+	if r.Type == InstancePlugin { // instance plugins are always first.
```

Comment:
what happens if there are multiple instance plugin

Reply:
There is only one instance but they will be initialized in registration order.

Reply:
I think we should disallow registering multiple ones at the moment.

Also, maybe s/InstancePlugin/RootPlugin/, but either is fine.